pkg/api/v1/types.go

```diff
@@ -1045,9 +1045,21 @@ func newPublicAnexiaCloudSpec(internal *kubermaticv1.AnexiaCloudSpec) (public *P
 type ClusterStatus struct {
 	// Version actual version of the kubernetes master components
 	Version ksemver.Semver `json:"version"`
-
 	// URL specifies the address at which the cluster is available
 	URL string `json:"url"`
+	// CCM represents the externalCCM status
+	CCM ExternalCCMStatus `json:"ccm"`
+}
+
+// ExternalCCMStatus represents the external CCM migration status
+// swagger:model ExternalCCMStatus
+type ExternalCCMStatus struct {
+	// the ClusterFeatureExternalCloudProvider in the cluster .spec.features
+	ExternalCCM bool `json:"externalCCM"`
+	// the logic AND of the CCMMigrationNeededAnnotation and CSIMigrationNeededAnnotation annotations
+	MigrationNeeded *bool `json:"migrationNeeded,omitempty"`
```

Comment:
This logic doesn't reflect is the migration needed. Instead, it reflects if the migration is in progress. To check if the migration is needed, you need to make sure that:

* ExternalCCM is `false`
* The provider supports the external CCM (using the function we already have)

The annotations are applied only when the migration has already started (by the mutating webhook).